### ISSUE TITLE
belongsTo:hasMany not persisting foreign key in 1.2.6

### ISSUE THREAD

**Issue by ryanberckmans** [firebase/emberfire#139]
In 1.2.6, new instances of a belongsTo:hasMany association aren't persisting the foreign key on the belongsTo side. Note the "double association" - Shift belongsTo twice.

My data model is as follows:

```
Employee hasMany Shift
Employer hasMany Shift
Shift belongsTo Employee
Shift belongsTo Employer
```

Works in 1.2.5:

```
// employer creates a newShift
1. employer, employee already exist
2. newShift = store.createRecord 'shift', { employer: employer }
3. newShift.save()
4. add newShift to employer.shifts
5. employer.save()

// employee is assigned newShift
6. newShift set to employee
7. newShift.save()
8. add newShift to employee.shifts
9. employee.save()
```

Shift belongsTo must be set before the Shift is added to the hasMany side.

When creating a new shift in 1.2.6, I must set both sides of the association locally, before any save(), or newShift.save() won't push the employer_id to Firebase.

Shift/Employer association works in 1.2.6:

```
// employer creates a newShift in 1.2.6
1. employer, employee already exist
2. newShift = store.createRecord 'shift', { employer: employer }
3. add newShift to employer.shifts # in 1.2.6, both sides must be set locally before any save()
4. newShift.save()
5. employer.save()
```

However, I couldn't get Shift/Employee association working in 1.2.6:

```
// employee is assigned newShift in 1.2.6
6. newShift set to employee
7. add newShift to employee.shifts # in 1.2.6, both sides must be set locally before any save()
8. newShift.save() # never sets newShift.employee in Firebase
9. employee.save() # correctly adds newShift to employee shifts collection in Firebase
```

In this Firebase update request for a newShift.save(), you can see that no "employee" key is set:

`r:0: update {"path":"/shifts/-JYkHBgb1wYy3Zfh1zz_","value":{"startDateAndTime":"2014-10-15T18:00:00.000Z","endDateAndTime":"2014-10-15T22:00:00.000Z","employer":"jim"}}`

Expected request:

`r:0: update {"path":"/shifts/-JYkHBgb1wYy3Zfh1zz_","value":{"startDateAndTime":"2014-10-15T18:00:00.000Z","endDateAndTime":"2014-10-15T22:00:00.000Z","employer":"jim","employee":"fred"}}`

Because newShift.employee is not being persisted to Firebase, I would expect the following bugs:
1. Two employees can point to the same shift
2. business.shifts[0].employee is undefined

Curiously, business.shifts[0].employee is in fact defined. When emberfire discovers a shift with undefined employee, it requests all employee records to clean up the dangling reference!

Response when requesting shift with undefined employee:

`handleServerMessage d {"p":"shifts/abc6","d":{"employer":"jim","endDateAndTime":"2015-05-04T16:00:00.000Z","startDateAndTime":"2015-05-04T13:00:00.000Z"}}`

Subsequent fetching of all employee shifts:

```
p:0: Listen on /spinners/fred for {}
p:0: Listen on /spinners/ryan for {}
```

Relevant response:

`handleServerMessage d {"p":"employees/fred","d":{"firstName":"Fred","lastName":"Smith","shifts":{"-JYkHBgb1wYy3Zfh1zz_":true,"abc1":true,"abc2":true,"abc4":true,"abc6":true}}}`

This handlebars template works, even though shift.employee is undefined:

```
{{#each business.shifts}}
  Employee: {{shift.employee}}
{{/each}
```

In conclusion, after failing to persist the shift.employee foreign key, emberfire is using the antipattern described in the [Firebase Structuring Data](https://www.firebase.com/docs/web/guide/structuring-data.html) guide.

> With this nested design, iterating the data becomes problematic. Even a simple operation like listing the names of rooms requires that the entire rooms tree, including all members and groups, be downloaded to the client

Help appreciated :neckbeard: . Meanwhile I'm using 1.2.5 and having ice cream for lunch.
